source/credentials_provider_profile.c

```diff
@@ -374,6 +382,23 @@ struct aws_credentials_provider *aws_credentials_provider_new_profile(
     struct aws_string *credentials_file_path = NULL;
     struct aws_string *config_file_path = NULL;
     struct aws_string *profile_name = NULL;
+    bool first_profile_in_chain = false;
+    if (source_profiles_table == NULL) {
+        source_profiles_table = aws_mem_calloc(allocator, 1, sizeof(struct aws_hash_table));
+        first_profile_in_chain = true;
+        /* source_profiles_table is an hashtable of (aws_byte_cursor *) -> NULL to detect recursion loop */
+        if (aws_hash_table_init(
+                source_profiles_table,
+                allocator,
+                3,
+                aws_hash_c_string,
+                aws_hash_callback_c_str_eq,
```

Comment:
comment says it's `(aws_byte_cursor *) -> NULL`. But based on hashing functions, seems like we're doing `(const char *) -> NULL`